extension/goplugin/sync.go

```diff
@@ -99,15 +103,27 @@ func (sync *Sync) Watch(ctx context.Context, path string, timeout time.Duration,
 
 // NewSync allocates Sync
 func NewSync(sync gohan_sync.Sync) *Sync {
-	return &Sync{raw: sync}
+	log.Debug("sync new: refs = %d", 1)
+	return &Sync{&syncRef{raw: sync, cnt: 1}}
 }
 
 // Clone allocates a clone of Sync; object may be nil
 func (sync *Sync) Clone() *Sync {
 	if sync == nil {
 		return nil
 	}
+	atomic.AddUint32(&sync.ref.cnt, 1)
+	log.Debug("sync clone: refs = %d", sync.ref.cnt)
 	return &Sync{
-		raw: sync.raw,
+		ref: sync.ref,
+	}
+}
+
+// Close closes Sync
+func (sync *Sync) Close() {
+	if atomic.AddUint32(&sync.ref.cnt, ^uint32(0)) == 0 {
+		sync.ref.raw.Close()
+		sync.ref.raw = nil
 	}
+	log.Debug("sync close: refs = %d", sync.ref.cnt)
```

Comment:
I believe these logs were useful when debugging this specific problem, but as soon as this code goes to production, I'm worried it could be very noisy.
Let's add proper tests (I can't see any in this PR) instead of polluting the log.